pkg/trace/api/otlp.go

```diff
@@ -187,6 +188,147 @@ func (o *OTLPReceiver) SetOTelAttributeTranslator(attrstrans *attributes.Transla
+	otelres := rspans.Resource()
+	resourceAttributes := otelres.Attributes()
+	resourceAttributesMap := make(map[string]string, resourceAttributes.Len())
+	resourceAttributes.Range(func(k string, v pcommon.Value) bool {
+		resourceAttributesMap[k] = v.AsString()
+		return true
+	})
+
+	// 1. First, attempt to get metadata from resource attributes
+	src, srcok := o.conf.OTLPReceiver.AttributesTranslator.ResourceToSource(ctx, rspans.Resource(), traceutil.SignalTypeSet)
+	if !srcok {
+		if v, ok := resourceAttributesMap["_dd.hostname"]; ok {
+			src = source.Source{Kind: source.HostnameKind, Identifier: v}
+			srcok = true
+		}
+	}
+	// TODO(songy23): use AttributeDeploymentEnvironmentName once collector version upgrade is unblocked
+	env := traceutil.GetOTelAttrVal(resourceAttributes, true, "deployment.environment.name", semconv.AttributeDeploymentEnvironment)
+	lang := traceutil.GetOTelAttrVal(resourceAttributes, true, semconv.AttributeTelemetrySDKLanguage)
+	if lang == "" {
+		lang = fastHeaderGet(httpHeader, header.Lang)
+	}
+	containerID := traceutil.GetOTelAttrVal(resourceAttributes, true, semconv.AttributeContainerID, semconv.AttributeK8SPodUID)
+	if containerID == "" {
+		containerID = o.cidProvider.GetContainerID(ctx, httpHeader)
+	}
+
+	// 2. Transform OTLP spans to DD spans. If metadata was missing in resource attributes, attempt to get it from spans
+	topLevelByKind := o.conf.HasFeature("enable_otlp_compute_top_level_by_span_kind")
+	ignoreResNames := make(map[string]struct{})
+	for _, resName := range o.conf.Ignore["resource"] {
+		ignoreResNames[resName] = struct{}{}
+	}
+	tracesByID := make(map[uint64]pb.Trace)
+	priorityByID := make(map[uint64]sampler.SamplingPriority)
+	var spancount int64
+	for j := 0; j < rspans.ScopeSpans().Len(); j++ {
+		libspans := rspans.ScopeSpans().At(j)
+		for k := 0; k < libspans.Spans().Len(); k++ {
+			otelspan := libspans.Spans().At(k)
+			if _, exists := ignoreResNames[traceutil.GetOTelResource(otelspan, otelres)]; exists {
+				continue
+			}
+
+			spancount++
+			traceID := traceutil.OTelTraceIDToUint64(otelspan.TraceID())
+			if tracesByID[traceID] == nil {
+				tracesByID[traceID] = pb.Trace{}
+			}
+			ddspan := transform.OtelSpanToDDSpan(otelspan, otelres, libspans.Scope(), topLevelByKind, o.conf, nil)
+			if !srcok {
+				src, srcok = traceutil.GetOtelSource(otelspan, otelres, o.conf.OTLPReceiver.AttributesTranslator)
+			}
+			if env == "" {
+				// no env at resource level, try the first span
+				if v := ddspan.Meta["env"]; v != "" {
+					env = v
+				}
+			}
+			if containerID == "" {
+				// no cid at resource level, grab what we can
+				_, containerID = transform.GetFirstFromMap(ddspan.Meta, semconv.AttributeContainerID, semconv.AttributeK8SPodUID)
+			}
+
+			if p, ok := ddspan.Metrics["_sampling_priority_v1"]; ok {
+				priorityByID[traceID] = sampler.SamplingPriority(p)
+			}
```

Comment:
env can be

Reply:
Is `_sampling_priority_v1` no longer supported? It gets read in collector [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/36b603da19a159fc909d98493786f7978fed252b/receiver/datadogreceiver/internal/translator/traces_translator.go#L109) and in the previous iteration of this agent code, but [it's just hardcoded to 1 in the backend.](https://github.com/DataDog/logs-backend/blob/420fc373f36c7514869325f004a391ab49f28c13/domains/event-platform/libs/raw-content-parsers/src/main/java/com/fsmatic/workload/processing/SpansOTLPProtobufPayloadParser.java#L582)

Reply:
Talked to Andrew about this - the user can specify `sampling.priority` for use with the [Probabilistic Sampling Processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/probabilisticsamplerprocessor/README.md), and seems that `_sampling_priority_v1` needs to be set as well so that trace agent can respect the sampling decision (e.g., if a user specifies `sampling.priority` to be "always keep" and they expect it to be kept by the Probabilistic Sampling Processor, we need to let trace agent know to keep it as well.

Reply:
I added a test for this case in my last commit